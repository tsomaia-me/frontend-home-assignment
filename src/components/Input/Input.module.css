.root {
    width: 100%;
    padding: 8px 16px;
    background: white;
    border: 1px solid #e0e0e0;
    border-radius: 8px;
    font-size: 21px;
}

.root:focus {
    outline: none;
}
