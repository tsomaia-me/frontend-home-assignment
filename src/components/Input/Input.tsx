import React from 'react'
import styles from './Input.module.css'

export interface InputProps extends React.ComponentProps<'input'> {
}

function Input(props: InputProps) {
  return (
    <input
      {...props}
      className={props.className ? `${props.className} ${styles.root}` : styles.root}
    />
  )
}

export default Input
