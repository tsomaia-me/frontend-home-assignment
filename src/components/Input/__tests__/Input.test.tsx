import React from 'react'
import renderer from 'react-test-renderer'
import Input from '../Input'

test('renders Input correctly', () => {
  const json = renderer.create(<Input/>).toJSON()
  expect(json).toMatchSnapshot()
})
