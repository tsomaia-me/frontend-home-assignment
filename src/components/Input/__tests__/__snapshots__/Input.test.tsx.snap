// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input correctly 1`] = `
<input
  className="root"
/>
`;
